worlds/tloz/__init__.py

```diff
@@ -87,6 +88,21 @@ class TLoZWorld(World):
         }
     }
 
+    location_name_groups = {
+        "Shops": shop_locations,
+        "Take Any": take_any_locations,
+        "Sword Caves": sword_cave_locations,
+        "Level 1": level_locations[0],
+        "Level 2": level_locations[1],
+        "Level 3": level_locations[2],
+        "Level 4": level_locations[3],
+        "Level 5": level_locations[4],
+        "Level 6": level_locations[5],
+        "Level 7": level_locations[6],
+        "Level 8": level_locations[7],
+        "Level 9": level_locations[8],
+    }
```

Comment:
This should be wrong typing since location_name_groups is defined as ClassVar[Dict[str, Set[str]]] = {}. Does pycharm not complain about this?
Should be set(shop_location), set(take_any_locations), etc.

Or do other games do the same thing and should we update the definition in AutoWorld.py? AutoWorld's `__new__` already converts it to `frozenset()` and `frozenset(set(list))` is quite a bit slower than `frozenset(list)`? Maybe @Berserker66  at this point.

Reply:
FWIW, PyCharm doesn't seem to complain, at least not by default, might be some setting I have to turn on somewhere

Reply:
mypy says
```python
worlds/tloz/__init__.py:92: error: Dict entry 0 has incompatible type "str": "list[str]"; expected "str": "set[str]"  [dict-item]
``` 
and similar

Reply:
Ah, it requires mypy. Well that explains it then, in which case there are many errors in that file

Reply:
I had no knowledge of what mypy even was until this conversation. Should that be a requirement if it's expected for people to follow it?

Regardless, I will update the location name groups to be sets when I get a chance.

Reply:
> Should that be a requirement if it's expected for people to follow it?

Pycharm does a lot of that already, but is a lot less strict and needs to operate faster (ignoring some things), so mypy is not strictly required unless you want to get typing 100% correct.
Using mypy gives you some level of confidence that you did not misuse an API or a special case (such as ignoring the `None` case for an `Optional[Item] `).

The question here is if the type hint in Autoworld is "good" (correct), because all code operating on it would work with a list rather than a set, and providing a list (as you did) would have had better performance.